vioquine/assignment1/Q2SingleLinkedList.java

```diff
@@ -0,0 +1,55 @@
+import java.util.AbstractMap;
+import java.util.Map;
+
+public class Q2SingleLinkedList {
+    public static void main(String... args) {
+        int listLength = 10;
+        Node<Integer> list = createLinkedList(0,listLength);
+        for(int k=0; k < listLength; k++){
+            System.out.println("k="+k+"\t -> "+getKthLastElement(list,k).toString());
+        }
+    }
+
+    /**
+     * Creates a new SingleLinkedList with count elements using recursion
+     * @param value content of the first node
+     * @param count Number of elements in the new SingleLinkedList
+     * @return new SingleLinkedList
+     */
+    private static Node<Integer> createLinkedList(int value, int count) {
```

Comment:
What will happen here if the count is 0?

Reply:
Oh I missed that one. With count=0 it would lead in an endless recursion. Same for a negativ count. I will fix that.